### ISSUE TITLE
wal files take too much space

### ISSUE THREAD

**Issue by ttys3** [GreptimeTeam/greptimedb#1816]
### What type of bug is this?

Performance issue

### What subsystems are affected?

Standalone mode

### What happened?

try to write about 100000000 rows into greptimedb, I did not finish it. as about 1/5 written, the space usage too much:

```shell
4.2G    /var/lib/greptimedb
```

```
mysql> select count(*)/10000 from system;
+--------------------------------+
| COUNT(UInt8(1)) / Int64(10000) |
+--------------------------------+
|                           2100 |
+--------------------------------+
1 row in set (15.76 sec)
```

```shell
--- /var/lib/greptimedb -----------------------------
    3.1 GiB [####################################] /wal
    1.1 GiB [############                        ] /data
  608.0 KiB [                                    ] /logs
    0.0   B [                                    ] /cluster
e   0.0   B [                                    ] /.tmp
```

```shell
--- /var/lib/greptimedb/wal -----------------------------------
                                                   /..
    1.0 GiB [####################################]  0000000000000003.raftlog
    1.0 GiB [################################### ]  0000000000000001.raftlog
    1.0 GiB [################################### ]  0000000000000002.raftlog
  104.0 MiB [###                                 ]  0000000000000004.raftlog
    4.0 KiB [                                    ]  0000000000000001.rewrite
```

and , check container logs, got lots of :

```
2023-06-23T16:54:27.671709Z ERROR table::predicate: Failed to create predicate for expr, error: Internal("The type of Timestamp(Millisecond, None) GtEq Timestamp(Second, None) of binary physical should be same")
2023-06-23T16:54:27.673491Z ERROR table::predicate: Failed to create predicate for expr, error: Internal("The type of Timestamp(Millisecond, None) GtEq Timestamp(Second, None) of binary physical should be same")
```

### What operating system did you use?

Fedora 38

### How can we reproduce the bug?

yes.  just use api and try to write 100000000 rows into greptimedb

```shell
docker run -d --name greptime -p 4000:4000 -p 4002:4002 \
-v "/var/lib/greptimedb:/tmp/greptimedb" \
greptime/greptimedb:0.4.0-20230621-nightly standalone start \
--http-addr 0.0.0.0:4000 \
--rpc-addr 0.0.0.0:4001 \
--mysql-addr 0.0.0.0:4002 \
--postgres-addr 0.0.0.0:4003 \
--prom-addr 0.0.0.0:4004 \
--opentsdb-addr 0.0.0.0:4242
```

**Comment by killme2008** [comment]
Yes, the default WAL parameters are too large, so we changed them in this PR last week https://github.com/GreptimeTeam/greptimedb/pull/1809

You can change the WAL configurations in config file:
```toml
[wal]
# WAL data directory
# dir = "/tmp/greptimedb/wal"
file_size = "256MB"
purge_threshold = "4GB"
```
Or by environment variables https://docs.greptime.com/user-guide/operations/configuration#environment-variable

But a simple method is to use [the latest 0.3.1 docker](https://hub.docker.com/layers/greptime/greptimedb/0.3.1/images/sha256-082c4f1f3fc95b89c1394b39aee239a583936e82dab376a7ad4ebabd26918b9a?context=explore) image that contains this PR.

> 2023-06-23T16:54:27.671709Z ERROR table::predicate: Failed to create predicate for expr, error: Internal("The type of > Timestamp(Millisecond, None) GtEq Timestamp(Second, None) of binary physical should be same")
> 2023-06-23T16:54:27.673491Z ERROR table::predicate: Failed to create predicate for expr, error: Internal("The type of > Timestamp(Millisecond, None) GtEq Timestamp(Second, None) of binary physical should be same")

Sorry for that, it's a known issue  https://github.com/GreptimeTeam/greptimedb/issues/992, it doesn't affect the runtime results but is really annoying, we'll fix it ASAP. @v0y4g3r 

Thanks for your report, please let me know if you have other issues.

**Comment by killme2008** [comment]
The 0.3.1 is the latest official release that was published 3 days ago manually.

https://github.com/GreptimeTeam/greptimedb/releases/tag/v0.3.1

The `v0.4.0-nightly-20230621` release note doesn't contains the  PR https://github.com/GreptimeTeam/greptimedb/pull/1809 which was published before `0.3.1`:

https://github.com/GreptimeTeam/greptimedb/releases/tag/v0.4.0-nightly-20230621

You can always find the changes in the release notes. 

 Our building system automatically generates nightly builds every week. However, due to a failed build last week, we manually released version `v0.4.0-nightly-20230621` as compensation and it's before PR #1809 .

**Comment by killme2008** [comment]
> the count is **very slow**, and uses lots of memory (**4GB**)
> 
> ```
> mysql> select count(*) from system;
> +-----------------+
> | COUNT(UInt8(1)) |
> +-----------------+
> |        99999997 |
> +-----------------+
> 1 row in set (3 min 26.26 sec)
> ```

Yes, the query performance only counts on MPP right now, we are implementing the smart indexing and adding more optimization rules in the next 0.4 release.
